src/rosecommon/include/itemdb.h

```diff
@@ -0,0 +1,84 @@
+// Copyright 2016 Chirstopher Torres (Raven), L3nn0x
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http ://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*!
+ * \file ItemDB
+ * \brief this file contains the \s ItemDatabase class which loads all of the base item data
+ *
+ * \author Raven
+ * \date August 2017
+ */
+#pragma once
+#include <vector>
+#include <map>
+
+#include "iserialize.h"
+#include "singleton.h"
+#include "osiroseDatabase.h"
+
+namespace RoseCommon
+{
+  /*!
+  * \class ItemDef
+  * \brief Describes an item.
+  * \sa
+  *
+  * \author Raven
+  * \date August 2016
+  */
+  struct ItemDef {
+    uint8_t type;
+    uint8_t subtype;
+    uint16_t id;
+    uint32_t buy_price;
+    uint32_t sell_price;
+    uint16_t weight;
+
+    /// Damage
+    uint16_t atk;
+    uint16_t def;
+    uint16_t range;
+    uint8_t slots;
+    uint8_t level;
+
+    // Model id
+    uint32_t view_id;
+
+    std::string name;
+    std::string desc;
+    std::string script;
+
+    ItemDef();
```

Comment:
You can use default member initializer instead of a constructor here.

Reply:
default member initializer? I'm using the constructor to force all values to known valid values. if you don't do this, the compiler can just use random garbage in memory.

Reply:
I was talking about the new mechanism that appeared in C++14:
struct Foo {
   int a = 42;
};

I feel like it's clearer to read, but that's personal state.